src/components/ThemeButton.tsx

```diff
@@ -0,0 +1,111 @@
+import Link from 'next/link';
+import { useState } from 'react';
+import { twMerge } from 'tailwind-merge';
+
+export default function ThemeButton({
+  href,
+  children,
+  className,
+  type = 'rectangle',
+  disabled = false,
+  noBackground = false,
+  fetchOnClick = false,
+  onClick = (_e: any) => {
+    /*action*/
+  },
+  onKeyDown = (_e: any) => {
+    /*action*/
+  },
+  onKeyUp = (_e: any) => {
+    /*action*/
+  },
+  onFetchDone = async (_response: any) => {
+    /*response*/
+  },
+}: any) {
+  const [fetchStatus, setFetchStatus] = useState(false);
+
+  const pillFormat = `${type == 'pill' ? 'rounded-full ' : ''}`;
+  const noBackgroundFormat = `${
+    disabled
+      ? `cursor-not-allowed${
+          noBackground ? ' ' : ' bg-black-0 '
+        }text-black-200 `
+      : `cursor-pointer${
+          noBackground ? ' text-accent-500' : ' bg-accent-500 text-secondary'
+        }`
+  }`;
+
+  function onClickInternal(e: any) {
```

Comment:
On internal links, I'd rather use something like [nProgress](https://github.com/rstacruz/nprogress) to indicate the loading vs a spinner

Reply:
The spinner is also taken from Figma.